lib/spore/environment.rb

```diff
@@ -17,11 +17,20 @@ def read
     end
 
     def apply
-      each { |k, v| ENV[k] ||= v }
+      each { |k, v| ENV[k] ||= maybe_indirect(v) }
     end
 
     def apply!
-      each { |k, v| ENV[k] = v }
+      each { |k, v| ENV[k] = maybe_indirect(v) }
+    end
+
+    private
+
+    # Indirect values point to other environment variables and start with two
+    # tildes (~~).
```

Comment:
I kind of made it up but yes, that's the intention. My thinking behind it is that allowing full ability to eval an arbitrary piece of Ruby or Shell is a potential security vulnerability and using $ as an indicator of another instance variable could yield false positives. So double tilde it is. Feel free to suggest an alternative, I don't really have a strong opinion on what it should be as long as I have this option ;)